source/common/http/async_client_impl.cc

```diff
@@ -264,6 +282,17 @@ void AsyncRequestImpl::onTrailers(HeaderMapPtr&& trailers) {
 }
 
 void AsyncRequestImpl::onReset() {
+  if (!cancelled_) {
+    // Add tags about reset.
+    child_span_->setTag(Tracing::Tags::get().Error, Tracing::Tags::get().True);
+    child_span_->setTag(Tracing::Tags::get().ErrorReason, "Reset");
+  }
+
+  // Finalize the span based on whether we received a response or not
+  Tracing::HttpTracerUtility::finalizeUpstreamSpan(
+      *child_span_, remoteClosed() ? &response_->headers() : nullptr,
+      remoteClosed() ? response_->trailers() : nullptr, streamInfo(), Tracing::EgressConfig::get());
```

Comment:
nit: you don't need to check remoteClosed() for the trailers, it will just be nullptr if it's nullptr

Reply:
Actually, this doesn't work because `this->response_` itself will be `nullptr`. This occurs in the cancellation flow.

```
[ RUN      ] AsyncClientImplTest.CancelRequest
source/common/http/async_client_impl.cc:294:26: runtime error: member call on null pointer of type 'struct MessageImpl'
```

Reply:
OK, thanks